src/extensions/ITemplateExtension.ts

```diff
@@ -0,0 +1,9 @@
+import { IExtensionUtilities } from './IExtensionUtilities';
+import { TemplateContext } from '../compilation';
+import { ScopeData } from '../compilation';
+import { XmlNode } from 'src';
+
+export interface ITemplateExtension {
+    getXmlDocuments(templateContext: TemplateContext): Promise<Map<string, XmlNode>>;
```

Comment:
Please remove this method.
This is quite specific to the current extensions. I think the extension api should be more general. I would leave only the `execute` method and have the extension author decide the internal logic himself.

Reply:
Agreed